### PR TITLE
[ADD] animal_owner

### PR DESCRIPTION
**Pull request by max3903**
Depends on #973

### CODE REVIEW

File: animal_owner/models/animal.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Copyright (C) 2020 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class Animal(models.Model):
+    _inherit = "animal"
+
+    partner_id = fields.Many2one("res.partner", string="Owner", index=True)
```
Comment (marcelsavegnago) [inline animal_owner/models/animal.py]: track_visibility is interesting

**Review by marcelsavegnago — COMMENTED** [review]
Some comments only

File: animal_owner/views/res_partner.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<odoo>
+
+    <!-- Partner Form View -->
+    <record id="view_partner_animal_owner_form" model="ir.ui.view">
+        <field name="name">view.partner.animal.owner.form</field>
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_partner_form"/>
+        <field name="arch" type="xml">
+            <notebook position="inside">
```
Comment (marcelsavegnago) [inline animal_owner/views/res_partner.xml]: Is it necessary to list animals in a notebook? I sincerely prefer it this way: (but it's a personal option)
https://github.com/odoo/odoo/blob/1527db8cd719c487ace02fca9fde21430e31d350/addons/sale_crm/views/crm_lead_views.xml#L21

**Review by marcelsavegnago — COMMENTED** [review]
I believe that a filter was missing to be applied when the partner has no animals.

image

image

File: animal_owner/models/res_partner.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    )
+
+    @api.multi
+    def action_view_animals(self):
+        action = self.env.ref("animal.action_animal").read()[0]
+        if self.animal_count > 1:
+            action["domain"] = [("id", "in", self.animal_ids.ids)]
+        elif self.animal_count == 1:
+            action["views"] = [(self.env.ref("animal.view_animal_form").id, "form")]
+            action["res_id"] = self.animal_ids.ids[0]
```
Comment (len-foss) [inline animal_owner/models/res_partner.py]: `if self.animal_count == 1: ... else` should rid you of the issue mentioned by Marcel.

**Review by len-foss — APPROVED** [review]
Code review: LGTM (besides one minor comment)

File: animal_owner/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Animal Owner",
+    "version": "12.0.1.0.0",
+    "license": "AGPL-3",
+    "summary": "Add owner to the animal",
+    "author": "Open Source Integrators, Odoo Community Association (OCA)",
+    "maintainer": "Open Source Integrators",
+    "website": "https://github.com/OCA/partner-contact",
+    "depends": ["animal", "contacts"],
```
Comment (len-foss) [inline animal_owner/__manifest__.py]: This does not depend on contracts?

**Review by marcelsavegnago — APPROVED** [review]
LGTM.. Do you intend to add tests?

**Comment by max3903** [comment]
@marcelsavegnago In a later pull request, once I get the end-users happy...

**Pull request merged**